User Data Handling Coding Exercise

umarmughal824 opened the pull request:
User Data Handling Coding Exercise

umarmughal824 committed:
add three fields (name, gender and education)

umarmughal824 committed:
add country field in the form

umarmughal824 committed:
fix dropdown value

InamTaj commented on src/modules/Lessons/8_LessonUserData/components/CheckBoxField.js:
```diff
@@ -0,0 +1,37 @@
+
+	onChangeHandler(event){
+		this.setState({
+			value: event.target.value
```
In the case of checkbox, we need to store all values which are checked. So you need to maintain an array instead of just single value.

InamTaj commented on src/modules/Lessons/8_LessonUserData/components/UserForm.js:
```diff
@@ -0,0 +1,85 @@
+		console.log(event.target.name+' = '+event.nativeEvent.target[index].text);
+
+		this.setState({
+			[event.target.name]: event.nativeEvent.target[index].text
```
why do you need to use `nativeEvent`? Just `event.target.value` should be enough for our usecase.

InamTaj commented on src/modules/Lessons/8_LessonUserData/components/UserForm.js:
```diff
@@ -0,0 +1,85 @@
+		console.log("Name: "+ this.state.name);
+		console.log("Gender: "+ this.state.gender);
+		console.log("education: "+ this.state.education);
+		console.log("Country: "+ this.state.country);
```
Please remove all these logs.

InamTaj commented on src/modules/common/components/CheckBox.js:
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
```
This is a duplicate file.

InamTaj commented on src/modules/common/components/Dropdown.js:
```diff
@@ -0,0 +1,44 @@
+	}
+
+	onChangeHandler(event){
+		var index = event.nativeEvent.target.selectedIndex;
```
please don't use `nativeEvent` since it's not required.

umarmughal824 committed:
code optimizations

umarmughal824 committed:
delete unreachable code

delete this file as this is no longer needed because its not being used anywhere in the project.

InamTaj commented on src/modules/Lessons/8_LessonUserData/components/UserForm.js:
```diff
@@ -0,0 +1,107 @@
+			radioButtonOptions: [{key:'male', value: 'Male'}, 
+			{key:'female', value: 'Female'}],
+			checkBoxOptions: [{key:'grad', value: 'Graduation'}, 
+			{key:'postgrad', value: 'Post Graduate'}],
```
the state can be improved such that you can provide default values of radio buttons or dropdowns by moving them in separate JS files

umarmughal824 committed:
move checkbox and radio buttons options to utils file

umarmughal824 committed:
Merge branch 'feature/UserForm' of https://github.com/umarmughal824/react-workshop into feature/UserForm

umarmughal824 committed:
add new line at the end of the file
